plugins/logitech-bulkcontroller/fu-plugin-logitech-bulkcontroller.c

```diff
@@ -0,0 +1,291 @@
+                                                              g_warning("Send data sync failed ID: %d ErrorCode: %d",
+                                                              id, error_code);
+        g_cond_signal(&user_data->test_upd_cond);
+}
+
+void
+bulk_file_transfer_cb (FileTransferState state, gint progress, BulkInterface bulk_intf, void *data)
+{
+        ApiUserData *user_data = (ApiUserData*) data;
+
+        switch (state) {
+                case TRANSFER_HASH_STARTED:
+                        g_debug("[%s] : File transfer hash in progress %u", user_data->prog_name, bulk_intf);
+                        break;
+                case TRANSFER_INIT_STARTED:
+                        g_debug("[%s] : File transfer init in progress %u", user_data->prog_name, bulk_intf);
+                        break;
+                case TRANSFER_STARTED:
+                        g_debug("[%s] : File transfer started for interface %u", user_data->prog_name, bulk_intf);
+                        break;
+                case TRANSFER_FAILED:
+                        g_warning("[%s] : File transfer failed for interface %u", user_data->prog_name, bulk_intf);
+                        g_cond_signal(&user_data->test_upd_cond);
+                        break;
+                case TRANSFER_INPROGRESS:
+                        if (BULK_INTERFACE_UPD == bulk_intf) {
+                                g_print ("\r[%s] : Interface: %u File uploaded : [%d%%]", user_data->prog_name, bulk_intf, progress);
+                                fflush(stdout);
+                                FuDevice *device = (FuDevice *)user_data->device_ptr;
+                                if (device)
+                                  fu_device_set_progress (device, progress);
+                        } else if (BULK_INTERFACE_SYNC == bulk_intf) {
+                                g_print("\r[%s] : Interface: %u File received : [%d%%]", user_data->prog_name, bulk_intf, progress);
+                                fflush(stdout);
+                        }
+                        if (progress == 100) {
+                                g_print("\n");
+                        }
+                        break;
+                case TRANSFER_COMPLETED:
+                        g_debug("[%s] :  File transfer completed for interface %u", user_data->prog_name, bulk_intf);
+                        g_cond_signal(&user_data->test_upd_cond);
+                        break;
+                default:
+                        break;
+        }
+}
+
+static gchar *
+get_device_version (LogiBulkController *obj, void *data) 
+{
+        Message message;
+        gint ret;
+        ReturnValue *ret_val;
+        ApiUserData *user_data = (ApiUserData*) data;
+        g_cond_init (&user_data->test_upd_cond);
+        g_mutex_init (&user_data->test_upd_mutex);
+        g_mutex_lock (&user_data->test_upd_mutex);
+        
+        memset(&message,'\0',sizeof(message));
```

Comment:
rather than doing this, please just do Message message = {0};

Reply:
Working on it